src/MultiMapManager/FlexMap.js

```diff
@@ -0,0 +1,47 @@
+import React, { useEffect } from 'react'
+import PropTypes from 'prop-types'
+import { connectToContext } from '../Provider/utils' // direct import required here!
+import { FlexMapStyled } from './styled'
+
+class FlexMap extends React.Component {
+  render () {
+    const { children, index, maps, numberOfColumns, numberOfRows, total, visibleState } = this.props
+    const totalMaps = total || maps.length
+    const visibleMapCount = visibleState.filter(_=>_).length
+    // TODO also check for index against total to see if its on a row with 1 or 2 collumns
+    const columns = numberOfColumns ||
+      (visibleMapCount === 1 || visibleMapCount === 3) && (index !== 0) ? 1 : 2
```

Comment:
`(visibleMapCount & 1 === 1) && (index !== 0) ? 1 : 2` would work for ANY odd number of visible maps